pylint/checkers/stdlib.py

```diff
@@ -599,6 +620,38 @@ def _check_lru_cache_decorators(self, decorators: nodes.Decorators) -> None:
                 confidence=interfaces.INFERENCE,
             )
 
+    def _check_dispatch_decorators(self, node: nodes.FunctionDef) -> None:
+        decorators_map: dict[str, tuple[nodes.NodeNG, interfaces.Confidence]] = {}
+
+        for decorator in node.decorators.nodes:
+            if isinstance(decorator, nodes.Name) and decorator.name:
+                decorators_map[decorator.name] = (decorator, interfaces.HIGH)
+            elif utils.is_registered_in_singledispatch_function(node):
+                decorators_map["singledispatch"] = (decorator, interfaces.INFERENCE)
+            elif utils.is_registered_in_singledispatchmethod_function(node):
+                decorators_map["singledispatchmethod"] = (
+                    decorator,
+                    interfaces.INFERENCE,
+                )
+
+        if "singledispatch" in decorators_map and "classmethod" in decorators_map:
+            decorator, confidence = decorators_map["singledispatch"]
+            self.add_message(
+                "singledispatch-method",
+                node=decorator,
+                confidence=confidence,
```

Comment:
Perhaps a bit pre-mature but let's skip the temporary assignment here. The code is short and clear enough that we can use `[0]` and `[1]`.